tortoise/tortoise.go

```diff
@@ -694,12 +695,16 @@ func (t *turtle) onAtx(atx *types.ActivationTxHeader) {
 			log.Uint64("weight", atx.GetWeight()),
 		)
 		epoch.atxs[atx.ID] = atx.GetWeight()
-		epoch.weight += atx.GetWeight()
+		if atx.GetWeight() > math.MaxInt64 {
+			// atx weight is not expected to overflow int64
+			panic("fixme: atx size overflows int64 " + strconv.FormatUint(atx.GetWeight(), 10))
+		}
+		epoch.weight = epoch.weight.Add(fixed.New64(int64(atx.GetWeight())))
```

Comment:
it is also expected that total epoch weight will not exceed uint64. does `fixed.New64` satisfy that?

Reply:
yes, it checks overflow on math operations